lightning/src/ln/channelmanager.rs

```diff
@@ -7097,23 +7138,28 @@ where
 			let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 			let peer_state = &mut *peer_state_lock;
 			let pending_msg_events = &mut peer_state.pending_msg_events;
-			peer_state.channel_by_id.retain(|_, chan| {
-				let retain = if chan.context.get_counterparty_node_id() == *counterparty_node_id {
-					if !chan.context.have_received_message() {
-						// If we created this (outbound) channel while we were disconnected from the
-						// peer we probably failed to send the open_channel message, which is now
-						// lost. We can't have had anything pending related to this channel, so we just
-						// drop it.
-						false
-					} else {
-						pending_msg_events.push(events::MessageSendEvent::SendChannelReestablish {
-							node_id: chan.context.get_counterparty_node_id(),
-							msg: chan.get_channel_reestablish(&self.logger),
-						});
-						true
+			peer_state.outbound_v1_channel_by_id.retain(|_, chan| {
+				if chan.context.get_counterparty_node_id() == *counterparty_node_id {
+					// If we created this (outbound) channel while we were disconnected from the
+					// peer we probably failed to send the open_channel message, which is now
+					// lost. We can't have had anything pending related to this channel, so we just
+					// drop it.
+					let retain = chan.context.have_received_message();
+					if !retain {
+						log_error!(self.logger, "Force-closing channel {}", log_bytes!(chan.context.channel_id()));
+						self.issue_channel_close_events(&chan.context, ClosureReason::HolderForceClosed);
+						self.finish_force_close_channel(chan.context.force_shutdown(false));
 					}
-				} else { true };
-				if retain && chan.context.get_counterparty_node_id() != *counterparty_node_id {
+					retain
+				} else { true }
+			});
+			peer_state.channel_by_id.iter_mut().for_each(|(_, chan)| {
+				if chan.context.get_counterparty_node_id() == *counterparty_node_id {
+					pending_msg_events.push(events::MessageSendEvent::SendChannelReestablish {
+						node_id: chan.context.get_counterparty_node_id(),
+						msg: chan.get_channel_reestablish(&self.logger),
+					});
+				} else {
```

Comment:
True!

However, I think I've come across something strange now in tests now where we get a mismatch between `chan.context.get_counterparty_node_id()` and the passed `counterparty_node_id` which sounds ridiculous (since we have channels per peer) so I'll check again with fresh eyes in the morning.

Reply:
Oh yeah, we're iterating over all the peer states with the containing for-loop. Why would we be doing that if we're only interested in one counterparty_node_id? Tests also break without the body in the else, which deals with channels that don't correspond to our passed in `counterparty_node_id`.